redis/redisLimiter.go

```diff
@@ -195,6 +196,72 @@ func (rl *rateLimiter) getFailConfig(mode Mode) (time.Duration, int64) {
 	return rl.freezeFailureConfig.limitPeriod, rl.freezeFailureConfig.maxFailures
 }
 
+func (rl *rateLimiter) getSecurityFailedTrials(ctx context.Context, key string) (uint64, error) {
+	failedTrials, err := rl.storer.GetValue(ctx, key)
+	if err != nil {
+		if err.Error() == "redis: nil" {
+			return 0, ErrKeyNotExists
+		}
+		return 0, err
+	}
+	convertedValue, err := strconv.ParseUint(failedTrials, 10, 64)
+	if err != nil {
+		return 0, err
+	}
+	return convertedValue, nil
+}
+
+// SetSecurityMode will set the key from volatile to persistent
+func (rl *rateLimiter) SetSecurityMode(key string) error {
+	ctx, cancel := context.WithTimeout(context.Background(), rl.operationTimeout)
+	defer cancel()
+
+	rl.mutStorer.Lock()
+	defer rl.mutStorer.Unlock()
+
+	wasSet, err := rl.storer.SetPersist(ctx, key)
+
+	if !wasSet {
+		return ErrKeyNotExists
+	}
+
+	if err != nil {
+		log.Error("Persist", "key", key, "err", err.Error())
+	}
+	return err
+}
+
+// UnsetSecurityMode will set the key from persistent to volatile
+func (rl *rateLimiter) UnsetSecurityMode(key string) error {
+	ctx, cancel := context.WithTimeout(context.Background(), rl.operationTimeout)
+	defer cancel()
+
+	rl.mutStorer.Lock()
+	defer rl.mutStorer.Unlock()
+
+	failedTrials, err := rl.getSecurityFailedTrials(ctx, key)
+	if err != nil {
+		return err
+	}
+	if failedTrials > 0 {
```

Comment:
I think that here the order of the calls should pe swapped. UnsetPersist should be called first, and then  we should verify the number of failedTrials and call a setExpire